### PR TITLE
Simplify InputMap

### PR DESCRIPTION
**Pull request by Shatur**
In this PR I make `InputMap` API more map-like. Also now it panics when something is going wrong instead of silently fails.

Should I replace `get` to return an iterator instead of `PetitiSet` directly?

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
> Should I replace get to return an iterator instead of PetitiSet directly?

No, I think that returning the `PetitSet` is more helpful to a more diverse collection of end users.

File: src/input_map.rs [hunk 1]
```diff
@@ -166,31 +164,39 @@ impl<A: Actionlike> InputMap<A> {
...
+    ///
+    /// If a matching input already existed in the set, it will be moved to the supplied index. Any input that was previously there will be moved to the matching input’s original index.
+    ///
+    /// # Panics
+    /// Panics if the map is full and `input` is not a duplicate or `input` is [`UserInput::Null`].
+    pub fn insert_at(&mut self, action: A, input: impl Into<UserInput>, index: usize) -> &mut Self {
+        let input = input.into();
+
+        // Don't insert Null inputs into the map
+        if input == UserInput::Null {
```
Comment (alice-i-cecile) [inline src/input_map.rs:195]: Can you please remove UserInput::Null in this PR? @sixfold correctly pointed out that this wasn't very useful.

File: src/input_map.rs [hunk 2]
```diff
@@ -221,7 +228,8 @@ impl<A: Actionlike> InputMap<A> {
     /// Any iterator that can be converted into a [`Button`] can be supplied, but will be converted into a [`PetitSet`] for storage and use.
     /// Chords can also be added with the [insert](Self::insert) method, if the [`UserInput::Chord`] variant is constructed explicitly.
     ///
-    /// Existing mappings for that action will not be overwritten.
+    /// # Panics
```
Comment (alice-i-cecile) [inline src/input_map.rs:231]:
```suggestion
    /// # Panics
    ///
```

File: src/input_map.rs (hunk 1 above)
Comment (Shatur) [reply src/input_map.rs:195]: You mean to remote it at all?

File: src/input_map.rs [hunk 3]
```diff
@@ -429,57 +348,18 @@ impl<A: Actionlike> InputMap<A> {
...
-
-    /// Returns how many bindings are currently registered for the provided action with the provided [`InputMode`]
-    ///
-    /// If `None` is provided, a total across all input modes will be provided.
-    ///
-    /// A maximum of `CAP` bindings across all input modes can be stored for each action,
-    /// and insert operations will silently fail if used when `CAP` bindings already exist.
+    /// Returns the `action` mappings
     #[must_use]
-    pub fn n_registered(&self, action: A, input_mode: Option<InputMode>) -> usize {
```
Comment (alice-i-cecile) [inline src/input_map.rs:471]: This API is still useful, especially to avoid unexpected panics.

File: src/input_map.rs (hunk 1 above)
Comment (Shatur) [reply src/input_map.rs:195]: If yes, how would you like to handle chords coertion if there is no inputs?

File: src/input_map.rs (hunk 3 above)
Comment (Shatur) [reply src/input_map.rs:471]: But maybe just use `get(action).len()`?

File: src/input_map.rs (hunk 1 above)
Comment (alice-i-cecile) [reply src/input_map.rs:195]: I meant remove it entirely, yes.

Which chord coercion in particular were you wondering about?

File: src/input_map.rs (hunk 3 above)
Comment (alice-i-cecile) [reply src/input_map.rs:471]: Yeah okay fair :)

File: src/input_map.rs (hunk 1 above)
Comment (Shatur) [reply src/input_map.rs:195]: I mean how to handle 0 in this case:

https://github.com/Leafwing-Studios/leafwing_input_manager/blob/2496c38d756275b7acedb19a5fc62b57a2632fbd/src/buttonlike_user_input.rs#L52-L56

Just create an empty chord?

Comment (alice-i-cecile) [reply src/input_map.rs:195]: Yeah, I think we should create an n-length chord in all cases there, rather than trying to be clever.

**Pull request merged**